sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -237,43 +320,65 @@ object AlluxioUtils extends Logging {
   }
 
   private def genFuncForAutoMountReplacement(
-      conf: RapidsConf,
       runtimeConf: RuntimeConfig,
       hadoopConf: Configuration,
-      alluxioBucketRegex: String) : Option[Path => Path] = {
+      alluxioBucketRegex: String): Option[Path => AlluxioReplacedFilePathInfoConvertTime] = {
     Some((f: Path) => {
       val pathStr = f.toString
-      if (pathStr.matches(alluxioBucketRegex)) {
-        initAlluxioInfo(conf)
+      val res = if (pathStr.matches(alluxioBucketRegex)) {
         val (access_key, secret_key) = getKeyAndSecret(hadoopConf, runtimeConf)
-
         val (scheme, bucket) = getSchemeAndBucketFromPath(pathStr)
         autoMountBucket(scheme, bucket, access_key, secret_key)
+        assert(alluxioMasterHost.isDefined)
+        AlluxioReplacedFilePathInfoConvertTime(
+          new Path(replaceSchemeWithAlluxio(pathStr, scheme, alluxioMasterHost.get)),
+          Some(scheme))
+      } else {
+        AlluxioReplacedFilePathInfoConvertTime(f, None)
+      }
+      logDebug(s"Automount replacing paths: $res")
+      res
+    })
+  }
 
-        // replace s3://foo/.. to alluxio://alluxioMasterHost/foo/...
-        val newPath = new Path(pathStr.replaceFirst(
-          scheme + ":/", "alluxio://" + alluxioMasterHost.get))
-        logDebug(s"Replace $pathStr to ${newPath.toString}")
-        newPath
+  // Contains the file string to read and contains a boolean indicating if the
+  // path was updated to an alluxio:// path.
+  case class AlluxioReplacedFilePathInfoTaskTime(fileStr: String, wasReplaced: Boolean)
```

Comment:
nit, could these two case classes names be shortened a bit?
`AlluxioPathReplaceTaskTime`
`AlluxioPathReplaceConvertTime` 

It may help with some of the longer lines that are needed.

Reply:
sure, I updated